### PR TITLE
Permutation of string added

### LINKED ISSUE #4728: Permutation of String
## 💥 Proposal

Hey, I wanted to add Permutation of string program in this ..
this basically tell you the maximum ways a string can be written..
Input: abc
Output:
abc
acb
bac
bca
cba
cab
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes, I have readed it..

### PR DESCRIPTION
**Pull request by UG-SEP**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes , I have 

### Description

I have added Permutation of string in c->other.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
closes: #4728

### CODE REVIEW

**Comment by UG-SEP** [comment]
@HarshCasper  please review it

**Comment by UG-SEP** [comment]
please review it guys @HarshCasper

File: C/other/Permutation_of_string.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#include <stdlib.h>
+//declare permutation and swap functions
+void permutation(char *,int,int);
+void swap(char *,char *);
+int main()
+{
+    char *s;
+    // dynamically creating string length
+    s=(char*)malloc(sizeof(char)*1000);
+    // getting string
```
Comment (siddharth25pandey) [inline C/other/Permutation_of_string.c:14]: Display a message in print form as what are you taking input from the user for better UI

File: C/other/Permutation_of_string.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    gets(s);
+    // changing size to the length of string+1 to store null at end
+    s=realloc(s,strlen(s)+1);
+	//calling permutation
+    permutation(s,0,strlen(s)-1);
```
Comment (siddharth25pandey) [inline C/other/Permutation_of_string.c:20]: Display a message in print form as what are you showing output to the user for better UI

File: C/other/Permutation_of_string.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+}
+/*Example:
+Input: abc
+Output: 1(abc)
+        2(acb)
+        3(bac)
+        4(bca)
+        5(cba)
+        6(cab)
+Time Complexity: O(n!)
```
Comment (siddharth25pandey) [inline C/other/Permutation_of_string.c]: Time Complexity is **Wrong**
Also, Include Space Complexity

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@UG-SEP Do the suggested changes

File: C/other/Permutation_of_string.c (hunk 2 above)
Comment (UG-SEP) [reply C/other/Permutation_of_string.c:20]: I didn't understand what to show can you tell me

Comment (siddharth25pandey) [reply C/other/Permutation_of_string.c:20]: You need to print message like this 
`The Permutation of the string are:`
Hope it makes sense. Do, same for input also.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
@UG-SEP Kindly make the changes and let me known on Discord once its fit to review back again.

File: C/other/Permutation_of_string.c (hunk 2 above)
Comment (UG-SEP) [reply C/other/Permutation_of_string.c:20]: ok thanks

File: C/other/Permutation_of_string.c (hunk 3 above)
Comment (UG-SEP) [reply C/other/Permutation_of_string.c]: I don't know to find  space complexity of program

**Comment by UG-SEP** [comment]
i didn't write the space complexity here because i don't know how to find space complexity but i will learn it soon

File: C/other/Permutation_of_string.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	//swapping done
+}
+/*Example:
+Input: abc
+Output: 1(abc)
+        2(acb)
+        3(bac)
+        4(bca)
+        5(cba)
+        6(cab)
```
Comment (siddharth25pandey) [inline C/other/Permutation_of_string.c]: Sample I/O is **not matching** with the execution of the Program
* Order of Printing message is missing in Sample I/O

File: C/other/Permutation_of_string.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Time Complexity: O(n*n!)
```
Comment (siddharth25pandey) [inline C/other/Permutation_of_string.c:67]: Add Space Complexity

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@UG-SEP Changes required

File: C/other/Permutation_of_string.c (hunk 5 above)
Comment (UG-SEP) [reply C/other/Permutation_of_string.c:67]: and don't know to find space complexity please tell me the space complexity of the program

Comment (siddharth25pandey) [reply C/other/Permutation_of_string.c:67]: It will be same as Time Complexity for this case

Comment (UG-SEP) [reply C/other/Permutation_of_string.c:67]: ok

**Review by siddharth25pandey — APPROVED** [review]
@UG-SEP LGTM 💯 Able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @UG-SEP for his code

**Comment by UG-SEP** [comment]
thanks a lot @siddharth25pandey you help me a lot by let me to learn the right way to contribute thanks a lot 
well i send you connection request on linkedin @siddharth25pandey  please check it

**Comment by siddharth25pandey** [comment]
> thanks a lot @siddharth25pandey you help me a lot by let me to learn the right way to contribute thanks a lot
> well i send you connection request on linkedin @siddharth25pandey please check it

@UG-SEP Yeah, will happy to connect 👍

**Pull request merged**